src/python-bindings/collector.cpp

```diff
@@ -14,6 +14,12 @@
 
 #include "module_lock.h"
 
+// http://docs.python.org/3/c-api/apiabiversion.html#apiabiversion
+#if PY_MAJOR_VERSION >= 3
+   #define PyInt_Check(op)  PyNumber_Check(op)
+   #define PyString_Check(op)  PyBytes_Check(op)
```

Comment:
It looks like `PyInt_Check` is only used in a single place ([here](https://github.com/htcondor/htcondor/blob/master/src/python-bindings/classad.cpp#L865-L866])), which is already in a python2-only block.  So it seems we can safely drop this macro.

`PyString_Check` is only defined in the python2 c-api, which has a note (in https://docs.python.org/2/c-api/string.html) saying:
> These functions have been renamed to PyBytes_* in Python 3.x. Unless otherwise noted, the PyBytes functions available in 3.x are aliased to their PyString_* equivalents to help porting.

To me that means `PyBytes_Check` and `PyString_Check` are intended to be aliases, so defining the macro that way doesn't strike me as so out-of-place... But yeah if you still don't like it, I can define a `PY_COMPAT_BYTES_CHECK`.